hw/ip/i2c/data/i2c_testplan.hjson

```diff
@@ -604,33 +612,328 @@
+            - The CONTROLLER_EVENTS.BUS_TIMEOUT field should be set upon timeout.
+            - Software can begin a new transaction from the halted state after clearing the
+              'CONTROLLER_EVENTS' CSR.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: controller_mode_bus_idle_delay
+      desc: '''
+            The MULTI_CONTROLLER_MONITOR_EN feature allows the DUT-controller to join an
+            active bus environment without stomping over other controllers and active
+            transactions.
+            One part of this is to ensure bus idle periods between transactions are respected.
+
+            Stimulus
+            - Configure DUT/Agent as Controller/Agent respectively.
+            - Populate FMTFIFO/RXFIFO for a Controller-Mode transaction.
+            - Configure HOST_TIMEOUT_CTRL with the number of idle cycles.
+            - Enable CTRL.MULTI_CONTROLLER_MONITOR_EN before enabling CTRL.ENABLEHOST
+
+            Checking
+            - Check that the DUT-Controller does not begin transmitting until at least
+              the number of idle cycles in HOST_TIMEOUT_CTRL has elapsed.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: loopback_test
+      desc: '''
+            I2C Controller and Target modes are capable of operating simultaneously, as a valid
+            multi-controller application has an endpoint attempting to start a transaction, losing
+            arbitration to another controller, then accepting the transaction as addressed to the
+            Target-Mode function within the same device.
+            A loopback test exercises a similar function.
+
+            Stimulus
+            - Disable any I2C agents.
+            - Set both CTRL.ENABLEHOST and CTRL.ENABLETARGET
+            - Drive a transaction that is addressed to one of the Target-configured addresses.
+
+            Checking
+            - That data integrity is maintained, and the transaction completes normally from the
+              perspective of both operating modes.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: multi_controller_clock_synchronization
+      desc: '''
+            The DUT-controller should observe the I2C clock synchronization procedure when transacting on
+            the bus. This allows the arbitration process to function as specified.
+
+            Testbench
+            - Common I2C bus, with the following connected devices:
+              - DUT
+              - x1 Agent-Controller
+              - x1 Agent-Target
+
+            Stimulus
+            - Configure both controllers to use different timing parameters
+            - Configure both controllers to begin a transaction, where:
+              - A significant portion of the data is identical near the start of the transaction.
+              - The latter data is random, so either may win arbitration.
```

Comment:
This is nice test to have, but if you wanted to short-circuit this, you could probably have a Target Agent that pulls the clock low a little early. Arbitration is technically a separate subject, though synchronization and arbitration loss may both be observed in the same transaction.

Up to you if you want the separate agent in the testbench for a more complete model, though. Compared to that short-circuited approach, it may actually be easier to think about and work with three modeled entities instead of a bunch of special-case "fault" injection code. Hm... but I haven't really thought about how yet another agent's set of monitors should work... Would that be a problem? :thinking: